### PR TITLE
Browserifyable 2

### PR DESCRIPTION
**Pull request by markandrus**
Hi, I updated @formula1's PR, #356. I also added a Karma test that uses karma-mocha to test the browserified version of tap. Eventually, I'd like to test the browserified version with an updated karma-tap (or a karma-node-tap), but there are still details to iron out there (e.g., just parse `console.log` like karma-tap, or expose a global TAP/Test instance and hook into its Readable Stream).

### CODE REVIEW

**Comment by markandrus** [comment]
I think Appveyor is failing for reasons unrelated to this PR.

File: lib/test.js [hunk 1]
```diff
@@ -40,6 +41,12 @@ const Promise = require('bluebird')
 const bindObj = require('bind-obj-methods')
 const cwd = process.cwd()
 
+// TODO: Update stack-utils to use capture-stack-trace, too.
```
Comment (markandrus) [inline lib/test.js:44]: @isaacs WDYT about updating stack-utils to use capture-stack-trace, too?

**Comment by isaacs** [comment]
Landed this (with some rebasing and updating of dependencies) on v12.1.

**Comment by isaacs** [comment]
Thanks!

**Pull request closed without merging**